### PR TITLE
Remove "Primary Contacts" from Contact Tab LHS

### PR DESCRIPTION
**Pull request by kennsippell**
# Description

Remove "Primary Contacts" from Contact Tab LHS
#5084 

Regarding removing "Primary Contacts" from RHS: I'd like to take a bit of time to double-check the database interactions and can either push in new PR or merge here.

To keep things simple at first, this creates a new `hydrateDataRecords` which defaults to true. I suspect we may want the interface to default to `false`, but looking for thoughts on that before I make any changes.

# Review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://github.com/medic/medic-docs/blob/master/development/style-guide.md), documented if necessary.
- [ ] Documented: Announced in Changes.md in plain English. Configuration and user documentation on [medic-docs](https://github.com/medic/medic-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in Changes.md.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by kennsippell** [comment]
@garethbowen Yep!

**Comment by kennsippell** [comment]
I agree with the goal to consolidate the logic in HydrateContactNames. That said, HydrateContactNames was unique in that it takes summaries as inputs instead of docs.

To remove it, we can a) update GetSummaries to use lineage or b) update lineage to operate on summary inputs.  Since HydrateContactNames relies on GetSummaries, `a` would likely create a circular dependency (GetSummaries > lineage > GetSummaries).  `b` might confuse lineage's "separation of concerns" - as it would require new lineage interfaces which operate on a second type of input.

For now, I've just defaulted `hydrateContactNames` to false and kept `HydrateContactNames`.

**Comment by garethbowen** [comment]
@kennsippell Sorry, travis didn't detect this commit so the build didn't run so I can't merge. I can't commit a file to kick off a build because it's in your repo. Can you commit a trivial change to get travis to pick this up?

**Comment by garethbowen** [comment]
@kennsippell Thanks! Now it looks like an e2e is legitimately failing for some reason...

**Comment by kennsippell** [comment]
@garethbowen This is the change to get the e2e test passing again... https://github.com/medic/medic-webapp/pull/5090/commits/d2e508e9687eeeb924efd68b0fbccff63abb165c. 

I still don't fully understand why, but the root cause is how we mutate the parameter of the function. As discussed, these are tricky investigations which can be avoided with a rule like https://eslint.org/docs/rules/no-param-reassign

File: webapp/src/js/controllers/reports.js [hunk 1]
```diff
@@ -197,9 +197,10 @@ angular
         });
     };
 
-    var query = function(options) {
-      options = options || {};
+    var query = function(opts) {
+      const options = Object.assign({}, opts);
       options.limit = options.limit || 50;
+      options.hydrateContactNames = 'hydrateContactNames' in options ? !!options.hydrateContactNames : true;
```
Comment (garethbowen) [inline webapp/src/js/controllers/reports.js]: This is quite hard to read. What about using `_.extend` as below?

I think the ideal solution would be to destructure `opts`, eg:

```js
const query = ({ limit=50, hydrateContactNames=true }={}) => {
```

This is easy to read and self documenting, however it does require working out all the possible options which isn't easy.

File: tests/e2e/sms-gateway.js [hunk 2]
```diff
@@ -317,7 +317,7 @@ describe('sms-gateway api', () => {
             done();
           });
         })
-        .catch(done);
+        .catch(done.fail);
```
[the remark this answers is not in the document]
Comment (kennsippell) [reply tests/e2e/sms-gateway.js]: I think that is true on mocha but untrue in jasmine2 ...

[the remark this answers is not in the document]
Comment (garethbowen) [reply tests/e2e/sms-gateway.js]: What about `done(new Error('foo'))`?

File: webapp/src/js/controllers/reports.js (hunk 1 above)
Comment (kennsippell) [reply webapp/src/js/controllers/reports.js]: For destructuring, options actually has many properties and we send it downstream to Search which takes even more properties. So we'd have to destructure and restructure etc. I went with the `extend` option which is a nice clean command to shallow copy and provide defaults.

File: tests/e2e/sms-gateway.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kennsippell) [reply tests/e2e/sms-gateway.js]: Passes

[the remark this answers is not in the document]
Comment (garethbowen) [reply tests/e2e/sms-gateway.js]: I'm shocked. That's terrible!

Can you do a quick scan to see if we're doing this anywhere else?

**Comment by kennsippell** [comment]
The reason the test was failing is because of a bug in reports.js Changes callback. If a report is inserted into the database and it is at the end of the list, it will not appear. For example a scenario with list [A, B] insert [C] will result in [A, B] but [A,B,C] is expected.

To fix, I'm changing
`query({ silent: true, limit: liveList.count() });`
to
`query({ silent: true, limit: Math.max(50, liveList.count()) });`

In the case of this test failure, the list length was 0 so no items could be inserted. The test failed when run as part of the suite because the previous test ended on the Reports page - so the page was updated via Changes. The bug did not previously manifest because of `options.limit = options.limit || 50;` which handles the case properly for lists of length 0 only.

File: tests/e2e/sms-gateway.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (kennsippell) [reply tests/e2e/sms-gateway.js]: Looking for this specific pattern `catch(done)` I don't see any other cases in the `tests/e2e` folder.

**Pull request merged**